src/wovnio/html/HtmlConverter.php

```diff
@@ -273,7 +273,14 @@ private function buildHrefLang($lang_code)
             $url = $this->headers->removeLang($url, $this->store->defaultLang());
             $url = Url::addLangCode($url, $this->store, $lang_code, $this->headers);
         } else {
-            if ($lang_code !== $this->store->defaultLang()) {
+            $customDomainLangs = $this->store->getCustomDomainLangs();
+            if ($customDomainLangs !== null) {
+                $customDomainLangHasSource = $this->store->getCustomDomainLangs()->hasSource($lang_code);
+            } else {
+                $customDomainLangHasSource = false;
+            }
+
+            if ($lang_code !== $this->store->defaultLang() || $customDomainLangHasSource) {
```

Comment:
Remind me again why we need this extra check? So `removeLang` now always returns the physical URL (which is what `$this->headers->urlKeepTrailingSlash;` is). But when generating the hreflang, we need to convert this back into the virtual URL that is desired for the browser.

Reply:
Yep you are correct in that reasoning.